### ISSUE TITLE
Stuck in an update loop

### ISSUE THREAD

**Issue by minj** [GrapheneOS/os-issue-tracker#2182]
Once the smartphone restarts to update I get the app optimization screen as usual. Once it boots properly, I get notified that the update failed:

> Failed to install update. <...> Update passed verification but installation failed <...> Already processing an update, cancel it first.

Now what? How does one cancel an update? I really don't want to reflash...

I think I may have delayed a reboot too long and got 2 of them piled up, or something

**Comment by minj** [comment]
Hmz.. I rebooted twice manually (i.e. without clicking on the update notification) with an installation in between, and both updates could be installed one after the other, doh

**Comment by thestinger** [comment]
This is just because there's a post-update step where new updates can't be installed yet and the update client isn't aware of it so it starts trying to install an update too early. This means you hadn't rebooted for days to boot up the update you had pending.

**Comment by mercertom** [comment]
Same problem. I rebooted a few hours before this, so it's not the case that I "hadn't rebooted for days to boot up the update [I] had pending". Is the solution to reboot again when you get these errors?

I think in my case it was caused by a cpu-hungry app running while the update process was finalizing, and it crashed the app, probably crashed the update process too.

**Comment by thestinger** [comment]
That isn't related to this and your hypothesis about what happened doesn't really make sense.

**Comment by mercertom** [comment]
Yes. I have the exact same error. Probably, almost certainly, from a very different cause.

**Comment by mercertom** [comment]
Is the solution to reboot again when you get these errors?

I've rebooted, but it's not finalizing.

**Comment by thestinger** [comment]
It is finalizing, you're not giving it enough time. It has nothing to do with this.

**Comment by thestinger** [comment]
You're interrupting the update process by rebooting. Let it update. Open a new issue if you have an issue.
